src/game.cpp

```diff
@@ -3397,11 +3399,14 @@ void Game::updateSound(f32 dtime)
 			      camera->getCameraNode()->getUpVector());
 	sound->setListenerGain(g_settings->getFloat("sound_volume"));
 
+	LocalPlayer *player = client->getEnv().getLocalPlayer();
 
 	//	Update sound maker
-	soundmaker->step(dtime);
+	if (player->makes_footstep_sound)
+		soundmaker->step(dtime);
 
-	LocalPlayer *player = client->getEnv().getLocalPlayer();
+	// Tell the sound maker that there are no footstep sounds
+	soundmaker->makes_footstep_sound = player->makes_footstep_sound;
```

Comment:
Technically, this should be before the invocation of `soundmaker->step(dtime)` because what if `makes_footstep_sound == false` from the beginning? Also, the first time you read an uninitialized variable. To fully prevent that I'd add a default value to the `SoundMaker`'s constructor.